sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -91,6 +91,28 @@ export async function executeAtomXmlOperation(
   return serializer.deserialize(response);
 }
 
+/**
+ * @internal
+ * @hidden
+ * The key value pairs having undefined/null as the values are removed recursively from the object in order to address the following issues
+ * - ATOM based management operations throw a "Bad Request" error if empty tags are included in the xml request body at top level.
+ * - At the inner levels, Service assigns the empty strings as values to the empty tags instead of throwing.
+ *
+ * @param {{ [key: string]: any }} resource
+ */
+export function sanitizeSerializableObject(resource: { [key: string]: any }) {
+  Object.keys(resource).forEach(function(property) {
+    if (resource[property] == undefined) {
+      delete resource[property];
+    } else if (
+      typeof resource[property] === "object" &&
+      resource[property].constructor === {}.constructor
```

Comment:
I haven't seen a check like this used before, what's the reason for this?

Reply:
This is to differentiate among the "object"s. This would filter the JSON objects and won't match any array or other kinds of objects.
```ts
function func(obj:any){
    console.log(typeof obj ==="object", obj.constructor==={}.constructor);
}

func({abc:1});            // true,  true 
func(["a","b"]);          // true,  false 
func([{"a":1},{"b":2}]);  // true,  false 
func(new Date());         // true,  false 
func(123);                // false,  false
```

Reply:
SOF References:
https://stackoverflow.com/a/11183002/4137356
https://stackoverflow.com/a/36675341/4137356

Reply:
That's probably worth putting in as a comment (or even just factoring this out into a small function with a comment).

Reply:
btw, does `{}.constructor` construct a new empty object every time it's called? Or is JS smart enough to treat that empty literal as a constant of some sort?

Reply:
I think it constructs a new empty object. But nobody references this object, so should be fine? (because the garbage collection takes care of it)

Or do you think there is a better way to do this?

Reply:
I was thinking you only need to create one, statically, unless there's some benefit to recreating it each time.

Reply:
Add a const `EMPTY_JSON_OBJECT_CONSTRUCTOR`, let me know if that's good/